src/wp-includes/user.php

```diff
@@ -4945,16 +4926,7 @@ function wp_validate_user_request_key( $request_id, $key ) {
 	$expiration_duration = (int) apply_filters( 'user_request_key_expiration', DAY_IN_SECONDS );
 	$expiration_time     = $key_request_time + $expiration_duration;
 
-	$algo = function_exists( 'hash' ) ? 'sha256' : 'sha1';
-
-	if ( str_starts_with( $saved_key, '$P$B' ) ) {
-		// Back-compat for old phpass hashes.
-		require_once ABSPATH . WPINC . '/class-phpass.php';
-
-		if ( ! ( new PasswordHash( 8, true ) )->CheckPassword( $key, $saved_key ) ) {
-			return new WP_Error( 'invalid_key', __( 'The confirmation key is invalid for this personal data request.' ) );
-		}
-	} elseif ( ! hash_equals( $saved_key, wp_hash( $key, 'auth', $algo ) ) ) {
+	if ( ! wp_verify_hashed_value( $key, $saved_key ) ) {
```

Comment:
It might still be worthwhile to pass `auth` (or whatever) to this function and feed it as the argument to `wp_salt()` to ensure domain-separation is possible.